src/sankoff_handler.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019-2022 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+
+#include "sankoff_handler.hpp"
+
+Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count> SankoffHandler::GenerateLeavesPartials(const size_t node_count, const size_t pattern_idx) {
+    auto site_partials = Eigen::Matrix<double, Eigen::Dynamic, state_count>();
+    site_partials.resize(node_count, state_count);
+    site_partials.setZero();
+    for (size_t seq = 0; seq < site_pattern_.TaxonCount(); seq++) {
+        auto site_val = site_pattern_.GetPatternSymbol(seq, pattern_idx);
+        if (site_val < state_count) {
+            for (size_t i = 0; i < state_count; i++) {
+                if (i != site_val) {
+                    site_partials(seq, i) = static_cast<double>(INT_MAX);

```

Comment:
I was at first confused at the mix of ints and doubles here, but I guess it makes sense because we don't want to overflow a double, but have a big number.

Reply:
In any case can we have this be a class-level const? Perhaps it could be constexpr? It shows up a bunch in the tests so let's change it there.